### PR TITLE
Fix react-native build

### PR DESCRIPTION
**Pull request by davimacedo**
Close https://github.com/parse-community/Parse-SDK-JS/issues/1092

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,10 +49,10 @@ For server-side applications or Node.js command line tools, include `'parse/node
 const Parse = require('parse/node');
 ```
 
-For React Native applications, include `'parse/react-native'`:
+For React Native applications, include `'parse/react-native.js'`:
 ```js
 // In a React Native application
-const Parse = require('parse/react-native');
+const Parse = require('parse/react-native.js');
```
Comment (davimacedo) [inline README.md]: @dplewis before you ask, it is weird, but it does not work on react-native if imported as `require('parse/react-native')`. It does import `require('parse/reactnative')` if we change the name of the file to `reactnative.js`. I don't know exactly. Probably some bug in their side for resolving files with `-`.

Comment (dplewis) [reply README.md]: How did you figure this out?

Comment (davimacedo) [reply README.md]: I used `npm link` to install the SDK in a test react native project that I built and I did some tests. In one of them, I added a `console.log` in the `react-native.js` file and it is only printing out when I import using `require('parse/react-native.js');`

Comment (dplewis) [reply README.md]: I make it a habit of adding file extensions in my projects. One of the shortcomings of node.

**Comment by davimacedo** [comment]
Tests are now passing. I had to upgrade mongodb in travis from 3.2 to 3.6 (same as parse-server) since 3.2 does not support collation which is a recent requirement of parse-server to run.

**Comment by davimacedo** [comment]
wait... it worked well in the browser simulator but not working in iOS. I will do some more tests and let you know.

**Comment by dplewis** [comment]
Oh I didn’t realize that 3.6 is the requirement

**Comment by davimacedo** [comment]
I believe that collation is now required in Parse Server since https://github.com/parse-community/parse-server/commit/fd0b535159877fa9bc74fb645169f56814c67c60

**Comment by davimacedo** [comment]
The problem now is https://github.com/back4app/Parse-SDK-JS/commit/684b9e1a16a3dd0e5530d04675e4a8852177d5d0#diff-fb0c08f76d82dd2379cda467e5d0dbe6

crypto-js is not available for react native project. I'm trying to figure out what to do.

**Comment by davimacedo** [comment]
@dplewis it looks good now. Can you please take a look? I tested in the device and it is working fine even using:

```
Parse.enableEncryptedUser();
Parse.secret= 'my Secrey Key';
```

File: package.json [hunk 2]
```diff
@@ -69,6 +70,7 @@
     "jest": "24.9.0",
     "jsdoc": "3.6.3",
     "jsdoc-babel": "0.5.0",
+    "metro-react-native-babel-preset": "^0.58.0",
```
Comment (dplewis) [inline package.json]: Can you pin the packages?

**Comment by davimacedo** [comment]
Sorry for the delay. I've just pinned the package.

**Pull request merged**